File: src/main/java/me/centralhardware/znatoki/telegram/statistic/telegram/callbackHandler/student/DeleteUserCallback.java
package me.centralhardware.znatoki.telegram.statistic.telegram.callbackHandler.student;

import lombok.RequiredArgsConstructor;
import me.centralhardware.znatoki.telegram.statistic.i18n.ErrorConstant;
import me.centralhardware.znatoki.telegram.statistic.i18n.MessageConstant;
import me.centralhardware.znatoki.telegram.statistic.service.ClientService;
import me.centralhardware.znatoki.telegram.statistic.telegram.callbackHandler.CallbackHandler;
import org.springframework.stereotype.Component;
import org.telegram.telegrambots.meta.api.objects.CallbackQuery;
import org.telegram.telegrambots.meta.api.objects.User;

@Component
@RequiredArgsConstructor
public class DeleteUserCallback extends CallbackHandler {

    private final ClientService clientService;

    @Override
    public void handle(CallbackQuery callbackQuery, User from, String data) {
        if (!telegramService.isAdmin(from.getId())) {
            sender.sendMessageFromResource(ErrorConstant.ACCESS_DENIED, from);
            return;
        }

        clientService.findById(Integer.parseInt(data.replace("/delete_user",""))).ifPresent(pupil -> {

            if (!pupil.getOrganizationId().equals(getZnatokiUser(from).organizationId())){
                sender.sendText("Доступ запрещен", from);
                return;
            }

            pupil.setDeleted(true);
            clientService.save(pupil);
            sender.sendMessageFromResource(MessageConstant.PUPIL_DELETED, from);
        });
    }

    @Override
    public boolean isAcceptable(String data) {
        return data.startsWith("/delete_user");
    }
}

File: src/main/kotlin/me/centralhardware/znatoki/telegram/statistic/telegram/callbackHandler/student/DeleteUserCallback.kt
package me.centralhardware.znatoki.telegram.statistic.telegram.callbackHandler.student

import dev.inmo.tgbotapi.extensions.api.send.sendMessage
import dev.inmo.tgbotapi.extensions.behaviour_builder.BehaviourContext
import dev.inmo.tgbotapi.extensions.behaviour_builder.triggers_handling.onDataCallbackQuery
import me.centralhardware.znatoki.telegram.statistic.mapper.StudentMapper

fun BehaviourContext.deleteUserCallback() = onDataCallbackQuery(Regex("delete_user\\d+$")) {
    val id = it.data.replace("delete_user", "").toInt()
    StudentMapper.delete(id)
    sendMessage(it.from, "Ученик удален")
}
